service/pkg/server/services.go

```diff
@@ -27,3 +31,56 @@ func registerServices() error {
 	}
 	return nil
 }
+
+func registerServiceDb(ctx context.Context, cfg db.Config, svc serviceregistry.Service, d *db.Client) error {
+	reg := svc.Registration
+	// If the service doesn't require a db client, return
+	if !reg.DB.Required {
+		return nil
+	}
+
+	// If the db client is already set, return
+	if d != nil {
+		return nil
+	}
+
+	// Conditionally set the db client if the service requires it
+	// Currently, we are dynamically registering namespaces and don't offer the ability to apply
+	// config at the NS layer. This poses a problem where services under a NS want to share a
+	// database connection.
+	// TODO: this should be reassessed with how we handle registering a single namespace
+
+	slog.Info("creating database client", slog.String("namespace", reg.Namespace))
+	// Make sure we only create a single db client per namespace
+	var err error
+	d, err = db.New(cfg,
```

Comment:
@jrschumacher Maybe I'm missing something but are you creating a new pool for each service registered? 

If that's the case I think we need to share the connection pool. Especially if each service is using the shared db config.  

We could easily overwhelm any connection limits because it would be max connections * the number of services registered. Or we need to capture service specific db config potentially.

Reply:
Yea I expected there would be performance issues as we move to a production environment. 

I think this could get complex. If we have separate database servers per service we'd need to d to make sure they have seperate pools right?

We should probably talk about what we're going to support day one vs where we'd like to be in a few months.

Reply:
> I think this could get complex. If we have separate database servers per service we'd need to d to make sure they have seperate pools right?

Yes that's how it would work. But would require db specific config for each service. 

The way I have been thinking about this is if you need that separation just run that one service separate so it can have a different db config and isolation. There are probably couple ways to tackle this so we should probably sync on this sometime today.